end_to_end/tpu/run_sft.sh

```diff
@@ -1,44 +1,72 @@
 #!/bin/bash
 
 :'
-
 # This script demonstrates a full end-to-end workflow for Supervised Fine-Tuning (SFT)
-# a pre-trained model using MaxText. The script fine-tunes a pre-trained model using 
-# the `sft_trainer` on the `HuggingFaceH4/ultrachat_200k` dataset.
-
-# Commands to run fine-tuning:
-  export HF_TOKEN=<Hugging Face access token>
-
-  export PRE_TRAINED_MODEL=llama3.1-8b
-
-  # Chat tokenizer for the model
-  export PRE_TRAINED_MODEL_TOKENIZER=meta-llama/Llama-3.1-8B-Instruct
-
-  # MaxText-compatible model checkpoint
-  export PRE_TRAINED_MODEL_CKPT_PATH=<gcs path for model checkpoint>
-
-  # Output directory to store run logs
-  export BASE_OUTPUT_DIRECTORY=<output directory>
-
-  # Number of fine-tuning steps to run
-  export STEPS=100
-
-  bash end_to_end/tpu/run_sft.sh
+# a pre-trained model using MaxText. The script fine-tunes a pre-trained model using
+# the `sft_trainer` on the `HuggingFaceH4/ultrachat_200k` dataset and produces a
+# fine-tuned model in the Hugging Face format.
+#
+# This script supports two scenarios:
+#
+# 1. **Run SFT on a Hugging Face Checkpoint**
+#    The script will automatically convert a Hugging Face checkpoint
+#    to a MaxText checkpoint, run SFT, and then convert the fine-tuned
+#    checkpoint back to the Hugging Face format.
+#
+# 2. **Run SFT on a MaxText Checkpoint**
+#    The script will run SFT on a pre-converted MaxText checkpoint and
+#    then convert the fine-tuned checkpoint back to the Hugging Face format.
```

Comment:
Just for my understanding, why does it need to be converted back to Hugging Face format if the user started with a MaxText checkpoint? Does the user always want HF format at the end? Or should this part be behind a flag?